controllers/homework.js

```diff
@@ -864,9 +865,9 @@ router.get('/:assignmentId', function(req, res, next) {
             // Abgabenübersicht anzeigen (Lehrer || publicSubmissions) -> weitere Daten berechnen
             if (!assignment.private
                 && ((assignment.teacherId == res.locals.currentUser._id
-                    || ((assignment.courseId||{}).teacherIds||[]).includes(res.locals.currentUser._id)
-                    || ((assignment.courseId||{}).substitutionIds||[]).includes(res.locals.currentUser._id))
-                && assignment.courseId != null || assignment.publicSubmissions)) {
+                    || ((assignment.courseId || {}).teacherIds || []).includes(res.locals.currentUser._id)
+                    || ((assignment.courseId || {}).substitutionIds || []).includes(res.locals.currentUser._id))
+                    && assignment.courseId != null || assignment.publicSubmissions)) {
```

Comment:
Der Grund warum auch Schüler in diese Ansicht kommen können ist `assignment.publicSubmissions`, welches vom Lehrer beim erstellen/bearbeiten der Aufgabe angegeben werden kann.